Dockerfiles/midstream/fix_etc_passwd.sh

```diff
@@ -1,25 +1,33 @@
-#!/bin/bash -e
+#!/usr/bin/bash -e
 main() {
     local username="${SSH_AUTH_USER:-default}"
     local home
     home=$(mk_home "$username")
     setup_passwd "$username" "$home"
+    if [ "$#" -eq 0 ]; then
+      echo "Running inside a openshift environment \n
+            No arguments to script detected.\n
+            sleeping indefintely."
+      exec sleep inf
+    fi
 }
+
 mk_home() {
     local username="${1:?}"
-    mkdir -p "/tmp/home/$username" && \
-    echo "/tmp/home/$username" && \
-    return 0
+    [[ -d /workDir ]] && [[ -w /workDir ]] && \
+        mkdir -p "/workDir/home/$username" && \
+        echo "/workDir/home/$username" && \
+        return 0
     mktemp --tmpdir -d "${username}.XXXXXX"
 }
+
 setup_passwd() {
     local username="${1:?}"
     local home="${2:?}"
-    local uid="$(id -u)"
-    local gid="${3:-0}"
+    local uid="$3"
+    local gid="${4:-0}"
     local passwd=/etc/passwd
-
-    ! getent passwd $uid 2>/dev/null || return 0
+    ! whoami 2>/dev/null || return 0
```

Comment:
-1 for `whoami`.
please return back to `getent passwd`

Reply:
midstream team wanted to have the script as it is without any changes. 
I had to revert it back.